### PR TITLE
WebGLRenderer: Use inline tone mapping only when rendering to screen.

### LINKED ISSUE #26346: OutputPass: shader does not compile if renderer.toneMapping is set
### Description

Setting, for example,
```js
renderer.toneMapping = THREE.ACESFilmicToneMapping;
```
will cause `OutputPass` shader to fail to compile.

### Reproduction steps

1. Add `renderer.toneMapping = THREE.ACESFilmicToneMapping` to an example using `OutputPass`.

### Code

```js
renderer.toneMapping = THREE.ACESFilmicToneMapping
```

### Live example

* [jsfiddle-latest-release](https://jsfiddle.net/g3atw6k5/)
* [jsfiddle-dev](https://jsfiddle.net/hjqw94c5/)

### Screenshots

_No response_

### Version

r154dev

### Device

_No response_

### Browser

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #26346.

**Description**

This PR ensures the inline tone mapping can only be used when rendering to screen. Similar to `WebGLRenderer.outputColorSpace`, `WebGLRenderer.toneMapping` has no effect when rendering to a render target anymore.

When tone mapping is required in context of RTT, `OutputPass`/`OutputShader` has to be used. This class uses now `RawShaderMaterial` which solves #26346. Meaning it compiles even when `WebGLRenderer.toneMapping` is set to something different than `THREE.NoToneMapping`.

In the next step, `OutputPass` can extract the tone mapping settings from the renderer.

This PR has to be noted in the migration guide.

### CODE REVIEW

**Comment by Mugen87** [comment]
This PR actually fixes `webgl_shaders_ocean` since previously the tone mapping was applied twice to the water (the beauty pass for the reflections _and_ when the water itself was rendered). Now it only happens once.

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1629,7 +1629,18 @@ class WebGLRenderer {
 			const morphTargets = !! geometry.morphAttributes.position;
 			const morphNormals = !! geometry.morphAttributes.normal;
 			const morphColors = !! geometry.morphAttributes.color;
-			const toneMapping = material.toneMapped ? _this.toneMapping : NoToneMapping;
+
+			let toneMapping = NoToneMapping;
+
+			if ( material.toneMapped ) {
+
+				if ( _currentRenderTarget === null || ( _currentRenderTarget !== null && _currentRenderTarget.isXRRenderTarget === true ) ) {
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:1637]: Without this additional check, tone mapping would not properly work in XR environments. It's essentially the same check like for `colorSpace`.

**Comment by drcmda** [comment]
this breaks postprocessing, and HDR workflow. for instance we would previously set materials to toneMapped=false and then go beyond RGB 0-1, which gets then picked up by the threshold set by the bloom pass. threshold=1 means nothing gets bloom except the materials that go over threshold.

now the toneMapped property has no effect and practically everything that blooms is now broken. what is the migration path?

**Comment by Mugen87** [comment]
> we noticed that this pr breaks postprocessing, and HDR workflow. 

To clarify, this change did not break post processing and HDR workflow in this repository.

> is there a migration path?

The idea is to apply tone mapping during the post processing workflow with a separate (or combined pass). The inline tone mapping is now kept in sync with the inline color space conversion which is only intended if you render directly to screen.

**Comment by Mugen87** [comment]
Considering your concrete use case you have to implement selective bloom differently now like any other selective FX technique. Meaning you render two beauty passes and apply the bloom to only one of them.

**Comment by mrdoob** [comment]
Hmm...

> for instance we would previously set materials to toneMapped=false and then go beyond RGB 0-1, which gets then picked up by the threshold set by the bloom pass.

Does this work?

```js
material.emissive.set( 1, 0, 0 );
material.emissiveIntensity = 100;
```

**Pull request merged**